test/snippet/core/type_traits/function_traits.cpp

```diff
@@ -0,0 +1,18 @@
+#include <functional>
+#include <string>
+
+#include <seqan3/core/type_traits/function.hpp>
+#include <seqan3/std/concepts>
+
+std::function my_caller = [] (size_t arg1, std::string & arg2)
```

Comment:
:nail_care: but it's easier to understand the function at first glance.
```suggestion
std::function my_caller = [] (size_t position, std::string & sequence)
```